libs/cms-translations/src/lib/intl.service.ts

```diff
@@ -1,29 +1,29 @@
 import { Injectable } from '@nestjs/common'
-import { createIntl, createIntlCache, IntlShape } from '@formatjs/intl'
+import {
+  createIntl,
+  createIntlCache,
+  IntlShape,
+  MessageDescriptor,
+} from '@formatjs/intl'
 import { logger } from '@island.is/logging'
 import { Locale } from '@island.is/shared/types'
-import { StaticText } from '@island.is/application/core'
 
 import { CmsTranslationsService } from './cms-translations.service'
 import type { FormatMessage } from '../types'
 
 export type NestIntl = Promise<IntlShape<string>>
 
-export const formatMessage = (intl: IntlShape<string>): FormatMessage => (
-  descriptor: StaticText,
-  values?: Record<string, unknown>,
-): string => {
-  if (!descriptor) {
-    logger.warn('No descriptor passed to formatMessage function.')
-    return ''
-  }
-
-  if (typeof descriptor === 'string') {
-    return descriptor
-  }
+export const formatMessage = (intl: IntlShape<string>) =>
```

Comment:
Do we really need to wrap the default `formatMessage` with this extra check? Are we not just making it worse to enforce code that is depending on the cms-translations to use a custom formatMessage?

Reply:
Good question. This check is to make the NestJS implementation consistent with the React implementation (in `@island.is/localization`). 

I think the string support was originally added to improve DX when dealing with strings/translations of different origins, eg as props in shared components. Some UIs don't need translations for some reason and instead use hard coded strings.

The `undefined` sugar was added in #6507 to simplify code like this: `message ? formatMessage(message) : undefined`.

Reply:
I believe this check should instead be done in the individual project them selfs rather than trying to support all different use-cases in this library. I am mostly concerned here about changing the return value to be optionally undefined.

I just wanted to raise my concern, let me know how you want to proceed?

Reply:
I don't have a strong opinion on this. Some devs liked the sugar enough to implement it. But I'll leave the cms-translation consistency change out of this PR at least. Worth discussing on slack someday.